src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+public class Storage {
+    protected Path filePath;
+    protected Ui ui;
+
+    /**
+     * Class constructor for Storage Class specifying the filepath
+     */
+    public Storage(Path filePath) {
+        this.filePath = filePath;
+        ui = new Ui();
+    }
+
+    /**
+     * Returns an ArrayList of tasks loaded from data source
+     *
+     * @return           an arraylist of tasks
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        boolean directoryExists = Files.exists(filePath);
+
+        try {
+            if (directoryExists) {
+                List<String> data = Files.readAllLines(filePath);
+                for (String str : data) {
+                    Task task = convertDataToTask(str);
+                    tasks.add(task);
+                }
+            }
+        } catch (IOException e) {
+            ui.showError(e.getMessage());
+        }
+
+        return tasks;
+    }
+
+    /**
+     * Saves the given task to the data source
+     *
+     * @param task the given task to save
+     */
+    public void save(Task task) {
+        boolean directoryExists = Files.exists(filePath);
+        try {
+            if (directoryExists) {
+                String newData = task.toData() + '\n';
+                Files.write(filePath, newData.getBytes(), StandardOpenOption.APPEND);
+            } else {
+                throw new DukeException("Data file doesn't exist");
+            }
+        } catch (IOException | DukeException e) {
+            ui.showError(e.getMessage());
+        }
+    }
+
+    /**
+     * Update the given task in the data source
+     *
+     * @param taskNumber the task number of the task
+     * @param task       the task given
+     * @param action     actions to perform to the task, eg: d-delete, m-modify
+     */
+    public void update(int taskNumber, Task task, String action) {

```

Comment:
Neat and interesting way of updating the datafile whenever relavant commands are executed! I had only thought of saving the tasks when the user exits the programme, 'confirming' their changes to the datafile, in a manner of speaking.